deltametrics/mask.py

```diff
@@ -48,51 +53,308 @@ def show(self, **kwargs):
+
+        # identify set of points in both the convex hull polygon and
+        # defined as points_to_test and put these binary points into seamap
+        In = np.array(list(map(lambda pt: polygon.contains(pt),
+                               points_to_test)))
+        Shallowsea_ = sea[In]
+        seamap = np.zeros(bordermap.shape)
+        flat_inds = list(map(lambda x: np.ravel_multi_index(x, seamap.shape),
+                             np.fliplr(Shallowsea_)))
+        seamap.flat[flat_inds] = 1
+        seamap[:3, :] = 0
+
+        # define other points as these 'Deepsea' points
+        Deepsea_ = sea[~In]
+        Deepsea = np.zeros((numviews+2, len(Deepsea_)))
+        Deepsea[:2, :] = np.flipud(Deepsea_.T)
+        Deepsea[-1, :] = 200.  # 200 is a background value for waves1s later
+
+        # define points for the shallow sea and the shoreborder
+        Shallowsea = np.array(np.where(seamap > 0.5))
+        shoreandborder = np.array(np.where(bordermap > 0.5))
+        c1 = len(Shallowsea[0])
+        maxtheta = np.zeros((numviews, c1))
+
+        # compute angle between each shallowsea and shoreborder point
+        for i in range(c1):
+
+            diff = shoreandborder - Shallowsea[:, i, np.newaxis]
+            x = diff[0]
+            y = diff[1]
+
+            angles = np.arctan2(x, y)
+            angles = np.sort(angles) * 180. / np.pi
+
+            dangles = angles[1:] - angles[:-1]
+            dangles = np.concatenate((dangles,
+                                      [360 - (angles.max() - angles.min())]))
+            dangles = np.sort(dangles)
+
+            maxtheta[:, i] = dangles[-numviews:]
+
+        # set up arrays for tracking the shore points and  their angles
+        allshore = np.array(np.where(edges > 0))
+        c3 = len(allshore[0])
+        maxthetashore = np.zeros((numviews, c3))
+
+        # get angles between the shore points and shoreborder points
+        for i in range(c3):
+
+            diff = shoreandborder - allshore[:, i, np.newaxis]
+            x = diff[0]
+            y = diff[1]
+
+            angles = np.arctan2(x, y)
+            angles = np.sort(angles) * 180. / np.pi
+
+            dangles = angles[1:] - angles[:-1]
+            dangles = np.concatenate((dangles,
+                                      [360 - (angles.max() - angles.min())]))
+            dangles = np.sort(dangles)
+
+            maxthetashore[:, i] = dangles[-numviews:]
+
+        # define the shoreangles and seaangles identified
+        shoreangles = np.vstack([allshore, maxthetashore])
+        seaangles = np.hstack([np.vstack([Shallowsea, maxtheta]), Deepsea])
+
+        return shoreangles, seaangles
+
+
+class ChannelMask(BaseMask, OAM):
+    """Identify a binary channel mask.
 
     A channel mask object, helps enforce valid masking of channels.
 
     Examples
     --------
-
     Initialize the channel mask
-        >>> cmsk = dm.mask.ChannelMask(arr)
+        >>> cmsk = dm.mask.ChannelMask(velocity, topo)
```

Comment:
this isn't completely clear. Add before this line:

```
>>> velocity = rcm8cube['velocity'][-1, :, :]
>>> topo = rcm8cube['eta'][-1, :, :]
```

Reply:
Updated this for all of the mask code examples.